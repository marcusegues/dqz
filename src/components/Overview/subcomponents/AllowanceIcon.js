// @flow
import React from 'react';
// $FlowFixMe
import { View, Text, StyleSheet } from 'react-native';
import { moderateScale } from '../../../styles/Scaling';

const ownStyles = StyleSheet.create({
  quantityContainer: {
    marginTop: 5,
    paddingHorizontal: 8,
    paddingVertical: 2,
    backgroundColor: '#EBF8DD',
    borderRadius: 3,
  },
  quantityText: {
    color: '#1A1A1A',
    fontSize: moderateScale(12),
    fontFamily: 'roboto_light',
  },
});

type AllowanceIconProps = {
  quantity: number | string,
  unit: string,
  text: string,
};

export const AllowanceIcon = ({ quantity, unit, text }: AllowanceIconProps) => (
  <View style={ownStyles.quantityContainer}>
    <Text style={ownStyles.quantityText}>{`${text} ${quantity} ${unit}`}</Text>
  </View>
);
