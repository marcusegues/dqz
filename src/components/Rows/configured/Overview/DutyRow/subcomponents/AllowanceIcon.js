// @flow
import React from 'react';
// $FlowFixMe
import { View, Text, StyleSheet } from 'react-native';
import { moderateScale } from '../../../../../../styles/Scaling';
import { MAIN_BLACK } from '../../../../../../styles/colors';
import { type } from '../../../../../../styles/fonts';

const ownStyles = StyleSheet.create({
  quantityContainer: {
    marginTop: 5,
    paddingHorizontal: 6,
    paddingVertical: 2,
    backgroundColor: '#EBF8DD',
  },
  quantityText: {
    color: MAIN_BLACK,
    fontSize: moderateScale(12),
    fontFamily: type.light,
  },
});

type AllowanceIconProps = {
  quantity: number | string,
  unit: string,
  text: string,
};

export const AllowanceIcon = ({ quantity, unit, text }: AllowanceIconProps) => (
  <View style={ownStyles.quantityContainer}>
    <Text style={ownStyles.quantityText}>{`${text} ${quantity} ${unit}`}</Text>
  </View>
);
