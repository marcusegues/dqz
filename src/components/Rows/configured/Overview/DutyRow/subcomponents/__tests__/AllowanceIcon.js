// @flow
import React from 'react';
// $FlowFixMe
import { Text } from 'react-native';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { AllowanceIcon } from '../AllowanceIcon';

jest.mock('react-i18next', () => ({
  translate: () => Component => props => <Component t={() => ''} {...props} />,
}));

describe('AllowanceIcon', () => {
  test('AllowanceIcon renders correctly', () => {
    const tree = renderer
      .create(<AllowanceIcon quantity={4} unit="kg" text="test" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('shallow renders and text matches to "test 4 kg" ', () => {
    const wrapper = shallow(
      <AllowanceIcon quantity={4} unit="kg" text="test" />
    ).dive();
    expect(
      wrapper
        .find(Text)
        .children()
        .text()
    ).toEqual('test 4 kg');
    expect(wrapper).toMatchSnapshot();
  });
});
