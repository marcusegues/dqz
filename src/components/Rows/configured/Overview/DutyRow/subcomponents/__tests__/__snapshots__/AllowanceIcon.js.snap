// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AllowanceIcon AllowanceIcon renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#EBF8DD",
      "marginTop": 5,
      "paddingHorizontal": 6,
      "paddingVertical": 2,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#1A1A1A",
        "fontFamily": "roboto_light",
        "fontSize": 18.857142857142858,
      }
    }
  >
    test 4 kg
  </Text>
</View>
`;

exports[`AllowanceIcon shallow renders and text matches to "test 4 kg"  1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <View
    style={
      Object {
        "backgroundColor": "#EBF8DD",
        "marginTop": 5,
        "paddingHorizontal": 6,
        "paddingVertical": 2,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#1A1A1A",
          "fontFamily": "roboto_light",
          "fontSize": 18.857142857142858,
        }
      }
    >
      test 4 kg
    </Text>
  </View>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#1A1A1A",
            "fontFamily": "roboto_light",
            "fontSize": 18.857142857142858,
          }
        }
      >
        test 4 kg
      </Text>,
      "style": Object {
        "backgroundColor": "#EBF8DD",
        "marginTop": 5,
        "paddingHorizontal": 6,
        "paddingVertical": 2,
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "accessible": true,
        "allowFontScaling": true,
        "children": "test 4 kg",
        "ellipsizeMode": "tail",
        "style": Object {
          "color": "#1A1A1A",
          "fontFamily": "roboto_light",
          "fontSize": 18.857142857142858,
        },
      },
      "ref": null,
      "rendered": "test 4 kg",
      "type": [Function],
    },
    "type": "View",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#1A1A1A",
              "fontFamily": "roboto_light",
              "fontSize": 18.857142857142858,
            }
          }
        >
          test 4 kg
        </Text>,
        "style": Object {
          "backgroundColor": "#EBF8DD",
          "marginTop": 5,
          "paddingHorizontal": 6,
          "paddingVertical": 2,
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "accessible": true,
          "allowFontScaling": true,
          "children": "test 4 kg",
          "ellipsizeMode": "tail",
          "style": Object {
            "color": "#1A1A1A",
            "fontFamily": "roboto_light",
            "fontSize": 18.857142857142858,
          },
        },
        "ref": null,
        "rendered": "test 4 kg",
        "type": [Function],
      },
      "type": "View",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
